src/Slick/Utils.hs

```diff
@@ -0,0 +1,36 @@
+module Slick.Utils
+  ( getDirectoryPaths
+  , fmapFmapOptDescr
+  ) where
+
+import           Control.Monad
+import           Data.List                  (intersperse, sortBy)
+import           Data.List                  as L (intersperse, sortBy, (\\))
+import           Data.Ord                   (comparing)
+import qualified Data.Text                  as T
+import           Data.Time
+import           Development.Shake
+import           Development.Shake.FilePath
+import           System.Console.GetOpt
+import           System.Directory
+import           System.Directory.Extra     (listFilesRecursive, removeFile)
+import           System.Environment
+
+------------------------------------------------------------------------------
+-- Helper functions
+
+-- | Given a list of extensions and directories,
+--   find all files that match, and return full paths.
+getDirectoryPaths :: [FilePath]         -- ^ file pattern like *.md
+                  -> [FilePath]         -- ^ directories to look at
+                  -> Action [FilePath]
+getDirectoryPaths extensions dirs =
```

Comment:
Do we need a whole different `Utils` module; would this not fit inside `Build` or `Slick`

Reply:
It's typical (at least for me) module where I push helper functions to find them quickly and if they might be used in multiple modules. I can move them to another module easily.